taskfile.yaml

```
@@ -540,6 +551,36 @@ tasks:
         . "{{.VENV_DIR}}/bin/activate"
         poetry build --format wheel
 
+  uv-component:
+    internal: true
+    label: "{{.COMPONENT}}"
+    vars:
+      VENV_DIR: "{{.G_BUILD_DIR}}/{{.COMPONENT}}/venv"
+    requires:
+      vars: ["COMPONENT"]
+    sources:
+      - "{{.G_BUILD_DIR}}/{{.COMPONENT}}-venv.md5"
+      - "{{.ROOT_DIR}}/requirements.txt"
+      - "{{.TASKFILE}}"
+      - "/etc/os-release"
+      - "**/*"
+      - exclude: "dist/**/*"
+    dir: "components/{{.COMPONENT}}"
+    generates:
+      - "dist/*.whl"
+    deps:
+      - task: "component-venv"
+        vars:
+          COMPONENT: "{{.COMPONENT}}"
+          OUTPUT_DIR: "{{.VENV_DIR}}"
+    cmds:
+      - task: "clean-python-component"
+        vars:
+          COMPONENT: "{{.COMPONENT}}"
+      - |-
+        . "{{.VENV_DIR}}/bin/activate"
+        UV_CACHE_DIR="{{.G_BUILD_DIR}}/uv-cache" uv build --wheel
```

```suggestion
  uv-component:
    internal: true
    label: "{{.COMPONENT}}"
    requires:
      vars: ["COMPONENT"]
    sources:
      - "{{.TASKFILE}}"
      - "/etc/os-release"
      - "**/*"
      - exclude: "dist/**/*"
    dir: "components/{{.COMPONENT}}"
    generates:
      - "dist/*.whl"
    cmds:
      - task: "clean-python-component"
        vars:
          COMPONENT: "{{.COMPONENT}}"
      - "uv build --wheel -o dist"
```
I don't think u need to enter a virtual environment to build the project when using uv (which is good since it provides a better isolation).
I updated this task to remove any virtual environment related setup.

---

This is a good catch, I copied the implementation of uv-component task from python-component task. Here is the code for python-component task:
```
  python-component:
    internal: true
    label: "{{.COMPONENT}}"
    vars:
      VENV_DIR: "{{.G_BUILD_DIR}}/{{.COMPONENT}}/venv"
    requires:
      vars: ["COMPONENT"]
    sources:
      - "{{.G_BUILD_DIR}}/{{.COMPONENT}}-venv.md5"
      - "{{.ROOT_DIR}}/requirements.txt"
      - "{{.TASKFILE}}"
      - "/etc/os-release"
      - "**/*"
      - exclude: "dist/**/*"
    dir: "components/{{.COMPONENT}}"
    generates:
      - "dist/*.whl"
    deps:
      - task: "component-venv"
        vars:
          COMPONENT: "{{.COMPONENT}}"
          OUTPUT_DIR: "{{.VENV_DIR}}"
    cmds:
      - task: "clean-python-component"
        vars:
          COMPONENT: "{{.COMPONENT}}"
      - |-
        . "{{.VENV_DIR}}/bin/activate"
        poetry build --format wheel
```

Following your logic, then we don't need to enter a virtual environment to build with poetry build as well. Therefore, my question is should we also update python-component as well? and should we do it at this pr?

---

Without UV_CACHE_DIR, you would have` error: failed to create directory `/.cache/uv`: Permission denied (os error 13)` during the package build in gh workflow.

The new commits adds UV_CACHE_DIR as an environmental variable.

---

- Let me check about the uv cache thing.
- If we want to update python-component, we should do it in a separate PR. It's always a good practice to keep the PR clean.

---

I suspect we might have the same issue in the integration test (as it's also using uv).
According to the official doc: https://docs.astral.sh/uv/concepts/cache/#cache-directory, we should probably specify the cache using `--cache-dir`, which is more explicit and thus more readable.
However, I think we should defer this to another PR. I need to investigate more about this cache management and side effect, as it might be reused by all uv components. For now, how about using `--no-cache` to disable caching?
@coderabbitai can u create a feature request for configuring uv cache properly in our taskfile?